functions/services/sendgrid/index.js

```diff
@@ -15,10 +11,19 @@ const addEmailToList = async (email, listId) => {
             ]
         }
     };
-    const [response] = await client.request(request);
+
+    const sendgridClient = getSendgridClient();
+    const [response] = await sendgridClient.request(request);
     return response;
 };
 
+function getSendgridClient() {
+    const client = require("@sendgrid/client");
+    client.setApiKey(process.env.SENDGRID_API_KEY);
+
+    return client;
+}
```

Comment:
Why do you call `getSendgridClient` each time `addEmailToList` called?

the function `getSendgridClient` is initialization function and should happen only once (I'm not familiar with _sendgrid_)

Reply:
I am not so familiar with it either. I mainly made this change to accommodate my testing approach (kinda backward I know lol). My approach wasn't working if the `require` initialized the client.

Be open to hearing other ideas on how to accommodate these tests :)

Reply:
You need to mock the module before you require it, because each test is different you can do something like this:

```js
beforeEach(() => {
    jest.resetModules();

    jest.mock("@sendgrid/client", () => yourmockModule);

    addEmailList = require("./").addEmailList;
 });
```

You mocked the library without giving it another implementation

Reply:
@rluvaton awesome! that makes total sense and works very well. Thanks for the suggestion